libevmasm/Assembly.cpp

```diff
@@ -971,6 +972,84 @@ void appendBigEndianUint16(bytes& _dest, ValueT _value)
 	assertThrow(_value <= 0xFFFF, AssemblyException, "");
 	appendBigEndian(_dest, 2, static_cast<size_t>(_value));
 }
+
+uint16_t calculateMaxStackHeight(AssemblyItems const& _items, uint16_t _args)
+{
+	static auto constexpr UNVISITED = std::numeric_limits<size_t>::max();
+
+	solAssert(!_items.empty());
+	uint16_t maxStackHeight = _args;
+	std::stack<size_t> worklist;
+	std::vector<size_t> stackHeights(_items.size(), UNVISITED);
```

Comment:
This seems to correspond to `stack_height_max` from the EOF spec, so maybe let's include the `max` bit in the name for clarity:

```suggestion
	std::vector<size_t> maxStackHeights(_items.size(), UNVISITED);
```

The naming here is generally getting a little ambiguous because we're dealing with a maximum over maximum values. Both `stackHeight` and `maxStackHeight` represent maximum stack height in some sense. Something like `currentMaxHeight` and `overallMaxHeight` might be clearer.

Reply:
Also:
- `s` -> `successor`
- `it` -> `jumpInstructionIt` (or just `jumpInstruction`)

Reply:
Regarding the `it`. It's rather `tagIt`. We are looking for tag associated with the `rjump`

Reply:
Right. I guess it just confirms that we do need a better names for those :)

Reply:
> ```c++
> 	std::vector<size_t> maxStackHeights(_items.size(), UNVISITED);
> ```

Can you apply the bit about `stackHeights` as well?

Reply:
done